Add io.github.pleromix.IceBox

pleromix opened the pull request:
<!-- ⚠️⚠️ Submission pull request MUST be made against the `new-pr` **base branch** ⚠️⚠️  -->

<!-- 💡 Go to the preview tab to click the links below 💡 -->

### Please confirm your submission meets all the criteria

<!-- 💡 Please replace each `[ ]` with `[X]` when the step is complete 💡 -->

- [X] IceBox can assist you in generating PDF files from your images. <!-- insert the description here -->
- [X] The domain used for the application ID is [controlled by the application developer(s)][appid-domain] and the [application id guidelines][appid] are followed.
- [X] I have read and followed all the [Submission requirements][reqs] and the [Submission guide][reqs2].
- [X] I have [built][build] and tested the submission locally.
- [X] I am an author/developer/upstream contributor to the project. If not, I contacted upstream developers about this submission. **Link:**
https://github.com/pleromix/IceBox
<!-- 💡 Mention any additional maintainers needed below 💡 -->

<!-- ⚠️⚠️ DO NOT modify anything below this line ⚠️⚠️  -->

[appid-domain]: https://docs.flathub.org/docs/for-app-authors/requirements/#control-over-domain-or-repository
[appid]: https://docs.flathub.org/docs/for-app-authors/requirements#application-id
[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[reqs2]: https://docs.flathub.org/docs/for-app-authors/submission
[build]: https://docs.flathub.org/docs/for-app-authors/submission/#before-submission

hfiguiere commented on io.github.pleromix.IceBox.desktop:
This file should be (submitted and managed) upstream.

pleromix replied:
Could you please explain how to do this?

hfiguiere commented on io.github.pleromix.IceBox.metainfo.xml:
This file should be (submitted and managed) upstream.

hfiguiere commented on io.github.pleromix.IceBox.metainfo.xml:
```diff
@@ -0,0 +1,59 @@
+
+    <url type="homepage">https://github.com/pleromix/IceBox</url>
+
+    <content_rating type="oars-1.0"/>
```
```suggestion
    <content_rating type="oars-1.1"/>

```

hfiguiere commented on io.github.pleromix.IceBox.metainfo.xml:
```diff
@@ -0,0 +1,59 @@
+
+    <screenshots>
+        <screenshot type="default">
+            <image>https://raw.githubusercontent.com/pleromix/icebox/main/screenshots/image_1.png</image>
```
screenshot should never link to the tip of a branch as it is not immutable.

hfiguiere commented on io.github.pleromix.IceBox.png:
remove this. the SVG is enough.

This file should be (submitted and managed) upstream.

hfiguiere commented on io.github.pleromix.IceBox.svg:
This file should be (submitted and managed) upstream.

hfiguiere commented on io.github.pleromix.IceBox.yaml:
```diff
@@ -0,0 +1,40 @@
+app-id: io.github.pleromix.IceBox
+runtime: org.freedesktop.Platform
+runtime-version: '23.08'
```
current is 24.08

hfiguiere commented on io.github.pleromix.IceBox.yaml:
```diff
@@ -0,0 +1,40 @@
+        url: https://github.com/pleromix/icebox.git
+        tag: 1.0.0
+      - type: archive
+        url: https://download.bell-sw.com/java/21.0.6+10/bellsoft-jre21.0.6+10-linux-amd64-full.tar.gz
```
this look like a JAVA runtime. Why not using the one on flathub. That would also work on aarch64.

pleromix replied:
I'm unsure if FlatHub's Java runtime would contain JavaFx libraries, But Bellsoft JRE does!

hfiguiere replied:
if it doesn't it should be added to this build.

hfiguiere commented:
bot, build io.github.pleromix.IceBox

pleromix commented:
I've changed the YAML file according to your instructions and stripped the JAR file and JavaFx libraries to reduce the final bundle, And also replaced the Bellsoft JRE with `org.freedesktop.Sdk.Extension.openjdk21`

hfiguiere commented on io.github.pleromix.IceBox.yaml:
```diff
@@ -0,0 +1,47 @@
+        path: io.github.pleromix.IceBox.desktop
+      - type: file
+        path: io.github.pleromix.IceBox.svg
+      - type: file
```
this should be `type: archive`

pleromix replied:
I changed it.

hfiguiere commented on io.github.pleromix.IceBox.yaml:
```diff
@@ -0,0 +1,47 @@
+      - type: file
+        path: io.github.pleromix.IceBox.svg
+      - type: file
+        url: https://download2.gluonhq.com/openjfx/21.0.7/openjfx-21.0.7_linux-x64_bin-sdk.zip
```
once again, what happen with the aarch64 package that is built?

pleromix replied:
I set following option for my app and this archive file:

```
only-arches:
  - x86_64
```

pleromix replied:
I'm currently working on it.

hfiguiere commented on strip-openjfx.sh:
all of this could just be in `cleanup`.

pleromix replied:
I removed `strip-openjfx.sh` and moved those commands to `cleanup`.

hfiguiere commented on io.github.pleromix.IceBox.yaml:
```diff
@@ -0,0 +1,47 @@
+        path: run.sh
+      - type: file
+        path: strip-openjfx.sh
+      - type: file
+        path: io.github.pleromix.IceBox.metainfo.xml
+      - type: file
+        path: io.github.pleromix.IceBox.desktop
+      - type: file
+        path: io.github.pleromix.IceBox.svg
```
neither of these three files exist.

pleromix replied:
Now YAML will fetch these files from `source` directory.

hfiguiere commented on io.github.pleromix.IceBox.yaml:
```diff
@@ -0,0 +1,47 @@
+      - mkdir -p /app/openjfx
+      - unzip openjfx-21*.zip
+      - install -Dm755 strip-openjfx.sh /app/openjfx/strip-openjfx.sh
+      - chmod +x /app/openjfx/strip-openjfx.sh
```
```suggestion
```

install does that for you,,

pleromix replied:
Removed.

hfiguiere commented on io.github.pleromix.IceBox.yaml:
```diff
@@ -0,0 +1,47 @@
+      - cp -r javafx-sdk-21*/lib/* /app/openjfx/
+      - install -Dm644 icebox.jar /app/icebox.jar
+      - install -Dm755 run.sh /app/bin/run.sh
+      - chmod +x /app/bin/run.sh
```
```suggestion
```

pleromix replied:
Removed.

bbhtt commented on io.github.pleromix.IceBox.yaml:
```diff
@@ -0,0 +1,59 @@
+  - org.freedesktop.Sdk.Extension.openjdk21
+finish-args:
+  - --share=ipc
+  - --filesystem=home
```
why does it need home?

pleromix replied:
It grants permissions to read and write files. Is that correct? 🤔

hfiguiere replied:
it grants unfettered permission to read/write the home directory, while a properly sandboxed application doesn't need that as it would use the document portal.

pleromix replied:
> it grants unfettered permission to read/write the home directory, while a properly sandboxed application doesn't need that as it would use the document portal.

If I remove it, will the app still function normally?

hfiguiere replied:
I do not know your app. It's up to you to test this.

pleromix replied:
> I do not know your app. It's up to you to test this.

It's not a good idea, I ran some tests, removing it could limit the JavaFx's FileChooser functionality and this could affect user-experience and file availability to the app.

bbhtt replied:
how about giving only xdg-pictures documents etc.?

pleromix replied:
> how about giving only xdg-pictures documents etc.?

I've replaced `- --filesystem=home` with these options:

```
- --filesystem=xdg-pictures
- --filesystem=xdg-documents
- --filesystem=xdg-download
```

And the result is (on Debian 12 bookworm x86_64):
![Screenshot from 2025-04-21 11-58-12](https://github.com/user-attachments/assets/bc42d1f5-ac29-47c5-9851-74b9ff25c36d)

pleromix commented:
When my app will be available on the FlatHub store?

bbhtt commented:
bot, build io.github.pleromix.IceBox

bbhtt commented:
/merge
